client/session.go

```diff
@@ -431,9 +431,7 @@ func (s *session) Write(namespace string, id string, t time.Time, value float64,
 		if err != nil {
 			atomic.AddInt32(&resultErrs, 1)
```

Comment:
The main reason the atomic ints are used extensively is to avoid having to use the write/read lock unless there absolutely is an error that must be read and returned.  In the 99% case where there is no error it avoids the use of any lock use whatsoever which is ideal, however it's definitely not extremely clean.